volatility/framework/plugins/linux/check_creds.py

```diff
@@ -0,0 +1,72 @@
+# Volatility
+
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+"""
+@author:       Matt Tressler
+@contact:      matthewtressler10@gmail.com
+"""
+
+import logging
+from typing import List
+
+from volatility.framework import interfaces, renderers, exceptions, constants, contexts, objects
+from volatility.framework.automagic import linux
+from volatility.framework.configuration import requirements
+from volatility.framework.interfaces import plugins
+from volatility.framework.layers import intel
+from volatility.plugins.linux import pslist
+
+vollog = logging.getLogger(__name__)
+
+class check_creds(interfaces.plugins.PluginInterface):
+    """Checks if any processes are sharing credential structures"""
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name='primary',
+                                                     description='Memory layer for the kernel',
+                                                     architectures=["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(
+                name="vmlinux", description="Linux kernel symbols"),
+
+            requirements.PluginRequirement(
+                name='pslist', plugin=pslist.PsList, version=(1, 0, 0))
+        ]
+
+    def _generator(self):
+        vmlinux = contexts.Module(
+            self.context, self.config['vmlinux'], self.config['primary'], 0)
+
+        type_task = self.context.symbol_space.get_type(self.config['vmlinux'] + constants.BANG +"task_struct")
+
+        if not type_task.has_member("cred"):
+            vollog.error("this command is not supported by this profile")
```

Comment:
Not sure whether error stop execution or not, I might be tempted to raise here, but up to you.  5:)

Reply:
It does seem like error doesn't exit... What would be the exception that should be thrown here? And should we change profile to symbol table? or OS version? or is leaving 'profile' okay?

Reply:
Errr, I'd prefer it were "symbol table` (and the sentence capitalized), but I didn't want to nitpick!  5:P  5:)  The exception I'd throw is that the structure's not valid so probably a TypeError (since we were expecting an object that had a "cred" member, and this doesn't)...

Reply:
ok cool, will get this fixed and standardized. This won't be the only plugin to need this code path / exception thrown, so wanted it settled.

Reply:
Do we want the type error to specifically say what the issue is? Like explicitly say it's the cred member we are missing or just a standard "This command is not supported by this symbol table"

Reply:
So grepping through the code for `TypeError` should give other examples.  `This command is not ...` doesn't help all that much, so something that tells someone what's gone wrong.  Something like "This version of the operating systems doesn't have the necessary structures for this plugin`, or even `Task requires a cred member` would both be fine.  Ideally errors are unique so we can easily trace back where they're thrown if they're reported to us...

Reply:
Makes sense. Just wanted to clarify before I made the changes.